lib/karafka/base_controller.rb

```diff
@@ -95,33 +98,12 @@ def params_batch=(messages)
       @params_batch = Karafka::Params::ParamsBatch.new(messages, topic.parser)
     end
 
-    # @return [Karafka::Params::Params] params instance for non batch processed controllers
-    # @raise [Karafka::Errors::ParamsMethodUnavailable] raised when we try to use params
-    #   method in a batch_processed controller
-    def params
-      raise Karafka::Errors::ParamsMethodUnavailable if topic.batch_processing
-      params_batch.first
-    end
-
     # Executes the default controller flow, runs callbacks and if not halted
-    # will schedule a call task in sidekiq
-    def schedule
-      run_callbacks :schedule do
-        case topic.processing_backend
-        when :inline then
-          call_inline
-        when :sidekiq then
-          call_async
-        else
-          raise Errors::InvalidProcessingBackend, topic.processing_backend
-        end
-      end
-    end
-
-    # @note We want to leave the #perform method as a public API, but just in case we will do some
-    #   pre or post processing we use call method instead of directly executing #perform
+    # will call process method of a proper backend
     def call
```

Comment:
Much cleaner interface here

Reply:
You know, that I did this mostly for your #147 ;) to simplify things (no rush with that one though)